vfs-azure/src/main/java/com/sludev/commons/vfs2/provider/azure/AzFileObject.java

```diff
@@ -512,136 +502,179 @@ protected boolean doSetLastModifiedTime(long modtime) throws Exception {
+
+            int blockSize = getBlockSize(srcFile.getContent().getSize());
+
+            ParallelTransferOptions opts =
+                    new ParallelTransferOptions(blockSize, 5, null);
+
+            BlobRequestConditions requestConditions = new BlobRequestConditions();
+
+            BlobOutputStream bos = destBlobClient.getBlockBlobClient().getBlobOutputStream(
+                    opts, null, null, null, requestConditions);
+
+            InputStream is = srcFile.getContent().getInputStream();
+
+            try {
+                byte[] buffer = new byte[4 * 1024 *1024];
+
+                for (int len; (len = is.read(buffer)) != -1; ) {
+                    bos.write(buffer, 0, len);
                 }
             }
+            finally {
+                is.close();
+                bos.close();
+            }
 
+            doGetType();
+        }
+        finally {
+            destFile.close();
+            srcFile.close();
         }
-
-        log.debug("Exit AZFileObject copy");
     }
 
 
     /**
-     * Returns the file CloudBlockBlob of the give file.
-     * If the file is not the type of AzFileObject or does not have its own CloudBlockBlob it is going to return the
-     * CloudBlockBlob of the current file.
+     * Performas a blob to blob copy for files larger and 256MB.
+     * @param srcFile
+     * @throws Exception
      */
-    private CloudBlockBlob getFileCurrBlob(FileObject destFile) {
+    private void doCopyFromUrl(AzFileObject srcFile) throws Exception {
 
-        CloudBlockBlob cloudBlockBlob = currBlob;
-        if (AzFileObject.class.isAssignableFrom(destFile.getClass())) {
+        BlockBlobClient srcBlobClient = srcFile.blobClient.getBlockBlobClient();
+        BlockBlobClient destBlobClient = blobClient.getBlockBlobClient();
 
-            cloudBlockBlob = ((AzFileObject) destFile).getCurrBlob();
-            if (cloudBlockBlob == null) {
+        // Get the list of committed blocks
+        BlockList blockList = srcBlobClient.listBlocks(BlockListType.COMMITTED);
+        List<Block> blocks = blockList.getCommittedBlocks();
 
-                cloudBlockBlob = currBlob;
-            }
+        long rangeMax = 0;
+        URL blobUrl = srcFile.getSignedURL(24);
 
+        List<String> blockIds = new ArrayList<>();
+
+        // For each block copy the block using a signed URL.
+        for (int j = 0; j < blocks.size(); j++) {
+
+            Block block = blocks.get(j);
+            long blockSize = block.getSize();
+            BlobRange range = new BlobRange(rangeMax, blockSize);
+            rangeMax += blockSize;
+
+            blockIds.add(block.getName());
+            destBlobClient.stageBlockFromUrl(block.getName(), blobUrl.toString(), range);
         }
 
-        return cloudBlockBlob;
+        destBlobClient.commitBlockList(blockIds, true);
+    }
+
+
+    protected int getBlockSize(long fileSize) throws FileSystemException {
```

Comment:
Worth a 2-line comment on this block size?

As I infer it, this is:
- use default 4MB block size, unless that's not big enough, in which case scale the block size with the overall file size to always use the maximum number of blocks